src/transmitter/index.ts

```diff
@@ -227,6 +244,7 @@ async function retryRequest(
   const options: NeedleOptions = {
     json: true,
     compressed: true,
+    agent,
   };
   if (config.HTTP_PROXY || config.HTTPS_PROXY) {
     options.agent = getProxyAgent(config, url);
```

Comment:
Will also need to ensure a keep-alive agent is used in the response to `getProxyAgent()` 🙏

Reply:
As discussed, not doing this for proxy connections yet